lib/recursive-methods.rb

```diff
@@ -1,27 +1,53 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+    if n == 0
+        return 1
+    else
+        return n * (factorial(n - 1))
+    end
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
```

Comment:
Time complexity is also O(n^2) since you have n recursive calls and each call creates a new array with `s[1..-1]`